Mage.Sets/src/mage/cards/w/WitheringWisps.java

```diff
@@ -0,0 +1,149 @@
+ */
+package mage.cards.w;
+
+import java.util.UUID;
+import mage.abilities.ActivatedAbilityImpl;
+import mage.abilities.TriggeredAbility;
+import mage.abilities.common.OnEventTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.condition.common.CreatureCountCondition;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.decorator.ConditionalTriggeredAbility;
+import mage.abilities.effects.common.DamageEverythingEffect;
+import mage.abilities.effects.common.SacrificeSourceEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.TargetController;
+import mage.constants.SuperType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.filter.FilterPermanent;
+import mage.filter.predicate.mageobject.SubtypePredicate;
+import mage.filter.predicate.mageobject.SupertypePredicate;
+import mage.filter.predicate.permanent.ControllerPredicate;
+import mage.util.CardUtil;
+
+/**
+ *
+ * @author L_J
+ */
+public class WitheringWisps extends CardImpl {
+
+    private static final String ruleText = "At the beginning of the end step, if no creatures are on the battlefield, sacrifice {this}.";
+
+    public WitheringWisps(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.ENCHANTMENT},"{1}{B}{B}");
+
+        // At the beginning of the end step, if no creatures are on the battlefield, sacrifice Withering Wisps.
+        TriggeredAbility triggered = new OnEventTriggeredAbility(GameEvent.EventType.END_TURN_STEP_PRE, "beginning of the end step", true, new SacrificeSourceEffect());
+        this.addAbility(new ConditionalTriggeredAbility(triggered, new CreatureCountCondition(0, TargetController.ANY), ruleText));
+
+        // {B}: Withering Wisps deals 1 damage to each creature and each player. Activate this ability no more times each turn than the number of snow Swamps you control.
+        this.addAbility(new WitheringWispsActivatedAbility());
+    }
+
+    public WitheringWisps(final WitheringWisps card) {
+        super(card);
+    }
+
+    @Override
+    public WitheringWisps copy() {
+        return new WitheringWisps(this);
+    }
+}
+
+class WitheringWispsActivatedAbility extends ActivatedAbilityImpl {
+
+    static class ActivationInfo {
+
+        public int turnNum;
+        public int activationCounter;
+
+        public ActivationInfo(int turnNum, int activationCounter) {
+            this.turnNum = turnNum;
+            this.activationCounter = activationCounter;
+        }
+    }
+
+    private static final FilterPermanent filter = new FilterPermanent("snow lands you control");
+    {
+        filter.add(new SupertypePredicate(SuperType.SNOW));
+        filter.add(new SubtypePredicate(SubType.SWAMP));
+        filter.add(new ControllerPredicate(TargetController.YOU));
+    }
```

Comment:
Unless I’m really misremembering how initilizars work, isn’t this modifying a static value in a non static way?

Reply:
Which looks fixed I another commit somewhere. Never mind.

Reply:
I already fixed it in 79d4c07